### PR TITLE
[Model] Add vision encoder and input embeddings merger warmup for Qwen2.5 VL model

### PR DESCRIPTION
**Pull request by kwang3939**
# Description
Add warm up of Qwen2.5 vl model vision encoder.
Pad the inputs to get_input_embeddings and warmup this embeddings merger.
Pack the multimodal funcs.

# Tests
CI: https://buildkite.com/tpu-commons/tpu-inference-ci/builds/4783

# Checklist

Before submitting this PR, please make sure:
- I have performed a self-review of my code.
- I have necessary comments in my code, particularly in hard-to-understand areas.
- I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

File: tpu_inference/models/jax/qwen2_5_vl.py [hunk 1]
```diff
@@ -1067,3 +1066,66 @@ def load_weights(self, rng_key: jax.Array) -> None:
...
+            )
+            dummy_grid_thw = grid_thw
+
+            run_compilation_fn("single_image_encoder",
+                               self.get_single_image_embedding,
+                               dummy_pixel_values,
+                               dummy_grid_thw,
+                               image_shape=input_hw)
+
+        for num_tokens in num_tokens_paddings:
```
Comment (hfan) [inline tpu_inference/models/jax/qwen2_5_vl.py]: Does this for loop have to be here instead of in the compilation_manager.py? It seems get_input_embedding (which calls the merger function) is available at the compilation_manager level already.

File: tpu_inference/runner/tpu_jax_runner.py [hunk 2]
```diff
@@ -730,8 +741,8 @@ def _get_input_ids_embeds(self, input_ids: jax.Array,
         if self.is_multimodal_model:
             inputs_embeds = self.get_input_embeddings_fn(
                 self.state,
-                input_ids=input_ids,
-                multimodal_embeddings=mm_embeds,
+                input_ids,
+                mm_embeds,
```
Comment (hfan) [inline tpu_inference/runner/tpu_jax_runner.py:745]: Is this change necessary?

Comment (kwang3939) [reply tpu_inference/runner/tpu_jax_runner.py:745]: Yes, I modified that to have compilation function to print num_tokens as **kwargs and compiled these two as *args. Otherwise, it would recompile later.

**Review by hfan — APPROVED** [review]
Thanks for this critical improvement!
